### DISCUSSION TITLE
Add config to sort file picker order

### DISCUSSION THREAD

**Discussion by tjmills-dev** [helix-editor/helix#9418] [General]
At the moment, the order of files in the file picker appears to be fairly random - directories are kept together but not much else. This makes it difficult to find files in larger directories.
It would be useful if they were:
- Sorted at all, perhaps by alphabetical order
- Had a configurable sort order, perhaps under editor.file_picker, for alphabet, file size, last modified, reverse, etc.

**Comment by tennox** [comment]
And it's not even directories kept together :cry:  - if I start a search:
![image](https://github.com/helix-editor/helix/assets/2084639/12cc2cf1-ed57-404d-91df-dd177983044f)

**Comment by pascalkuthe** [comment]
File picker is already sorted there was a pr for that a long time ago and we forgot to close this.

As soon as you start typing is sorted by score ofcourse and the sorting is only used as a tiebreaker

**Reply by tennox** [reply]
But in my example the tiebreaker doesn't seem to work... (I don't see the score being different)

**Reply by pascalkuthe** [reply]
Shorter matches are preffered.


It's score -> len -> alpabetix sorting (tie breaker).


Works exactly the same as fzf

**Comment by edrex** [comment]
IMO the file picker should sort files by -modtime. That would make the default view useful.

**Reply by edrex** [reply]
It seems that https://github.com/helix-editor/helix/pull/336 implements date sorting (presumably reverse?), but currently the file picker seems to be sorted by name. :confused:

**Comment by zmc** [comment]
Found this discussion while trying to figure out what the sort order even was, because I couldn't make any sense of it. Forcing the file picker to only sort by this odd combination of timestamps guarantees a really unhelpful sort order in some of the repos I work on, since files get created and modified constantly by people who aren't me. Combined with long paths, I sometimes have to read each and every line to find what I'm looking for.

Being able to sort by name would save me considerable time.

**Comment by shepelevstas** [comment]
Is it possible within same pathname length to sort search results by filename?

![image](https://github.com/user-attachments/assets/9d653582-4f29-4165-8683-87374274bc95)

It is much easier if results are groupd by files:

![image](https://github.com/user-attachments/assets/dad0e129-d658-46bc-8772-7c6422997dec)

**Comment by ximon18** [comment]
Rather than a config file setting, I would find it helpful to, similar to `%` handling, to have the ability to type something in the input to influence the sort order so that I can choose per picker invocation what I want to sort by.
